app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/SearchPeopleScreen.kt

```diff
@@ -0,0 +1,536 @@
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .height(if (searchPeopleScreenState.federatedBackendResultsCollapsed) fullHeight else dimensions().defaultSearchLazyColumnHeight)
+                            .animateItemPlacement()
+                    )
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun SearchResultContent(
+    headerTitle: String,
+    totalSearchResultCount: String,
+    searchResult: @Composable () -> Unit,
+    onShowAllClicked: () -> Unit,
+    onShowLessClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    ConstraintLayout(modifier) {
+        val (headerRef, columnRef, buttonRef) = createRefs()
+
+        Box(
+            Modifier
+                .fillMaxWidth()
+                .wrapContentHeight()
+                .constrainAs(headerRef) {
+                    top.linkTo(parent.top)
+                    bottom.linkTo(columnRef.top)
+                }
+        ) {
+            FolderHeader(name = headerTitle)
+        }
+        Box(
+            Modifier
+                .fillMaxWidth()
+                .wrapContentHeight()
+                .constrainAs(columnRef) {
+                    top.linkTo(headerRef.bottom)
+                    bottom.linkTo(buttonRef.top)
+
+                    height = Dimension.fillToConstraints
+                }) {
+            searchResult()
+        }
+        Box(
+            Modifier
+                .fillMaxWidth()
+                .wrapContentHeight()
+                .constrainAs(buttonRef) {
+                    top.linkTo(columnRef.bottom)
+                    bottom.linkTo(parent.bottom)
+                }
+        ) {
+            ShowButton(
+                totalSearchResultCount = totalSearchResultCount,
+                onShowAllClicked = onShowAllClicked,
+                onShowLessClicked = onShowLessClicked,
+                modifier = Modifier
+                    .align(Alignment.BottomEnd)
+                    .padding(end = dimensions().spacing8x)
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun ShowButton(
+    totalSearchResultCount: String,
+    onShowAllClicked: () -> Unit,
+    onShowLessClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    var isShowAll by remember { mutableStateOf(true) }
+
+    Box(modifier) {
+        AnimatedContent(isShowAll) { showAll ->
+            WireSecondaryButton(
+                text = if (showAll) "Show All ($totalSearchResultCount)" else "Show Less",
+                onClick = {
+                    if (isShowAll) onShowAllClicked() else onShowLessClicked()
+
+                    isShowAll = !isShowAll
+                },
+                minHeight = dimensions().showAllCollapseButtonMinHeight,
+                fillMaxWidth = false,
+            )
+        }
+    }
+}
+
+@Composable
+fun ExternalContactResultItem(
```

Comment:
File is getting too big, hard to find stuff.

Perhaps this `ExternalContactResultItem` and `ContactSearchResultItem` could be extracted to other files?


They don't depend on anything else. They could get their own PR to make this one smaller actually.